.github/workflows/go.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and Test
+
+on:
+  push:
+  
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.18
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Run Revive Action
+      uses: docker://morphy/revive-action:v2
+      with:
+        config: revive.toml
```

Comment:
Is Revive meant to be used on top of all the go linting tools that run as part of the precommit flow? 

Should we maybe add Revive to the pre-commit command as well so we don't end up with possibly conflicting linter rules between the CI and pre-commit hooks?

Reply:
Honestly I was just copyng and pasting from other repos, we can discuss this on monday, but yes we should just use one tool

Reply:
Cool SG